queries/atm.py

```diff
@@ -0,0 +1,571 @@
+            return " ".join(word_list[:-1]) + " " + number
+        return " ".join(word_list[:-1]) + " " + number_to_text(last_word, case="þf")
+    return s
+
+
+def _get_foreign_exchange_string(atm_list: List[Dict[str, Any]]) -> str:
+    """Return a string with foreign exchange info for the given ATM"""
+    atm: Dict[str, Any] = {}
+    # if len(atm_list) == 1:
+    atm = atm_list[0]
+    # else:
+    #    for a in atm_list:
+    #        if a["services"]["foreign_exchange"]["active"]:
+    #            atm = a
+    #            break
+    currency_abr = atm["services"]["foreign_exchange"]["currency"]
+    # Convert currency tags to strings through _FOREIGN_CURRENCY map
+    currencies: List[str] = list()
+    for i in range(len(currency_abr)):
+        print("abr: ", currency_abr[i])
+        currency = NounPhrase(_FOREIGN_CURRENCY[currency_abr[i].strip()]).accusative
+        print("currency: ", currency)
+        if currency is not None:
+            currencies.append(currency)
+    return natlang_seq(currencies)
+
+
+_ERRMSG = "Ekki tókst að sækja upplýsingar um hraðbanka."
+
+
+def _answ_for_atm_query(q: Query, result: Result) -> AnswerTuple:
+    """Return an answer tuple for the given ATM query"""
+    ans_start = ""
+    atm_list: List[Dict[str, Any]] = []
+    if "context_reference" in result and result.last_atm is not None:
+        # There is a reference to a previous result
+        # TODO: Handle multiple ATMs in same location, but with different services
+        atm_list = result.last_atm
+        atm_word = "hraðbankanum"  # if len(atm_list) == 1 else "hraðbönkunum"
+        if result.qkey == "AtmFurtherInfoDeposit":
+            # if any(atm["services"]["deposit"] is True for atm in atm_list):
+            if atm_list[0]["services"]["deposit"] is True:
+                ans_start = (
+                    f"Já, hægt er að leggja inn á reikninginn þinn í {atm_word} við "
+                )
+            else:
+                ans_start = f"Nei, ekki er hægt að leggja inn á reikninginn þinn í {atm_word} við "
+        elif result.qkey == "AtmFurtherInfoWithdrawalLimit":
+            ans_start = f"Hámarksúttekt í {atm_word} við "
+
+    elif "error_context_reference" in result:
+        return gen_answer("Ég veit ekki til hvaða hraðbanka þú vísar.")
+    else:
+        location = q.location
+        if location is None:
+            return gen_answer("Ég veit ekki hvar þú ert.")
+
+        if result.qkey == "ClosestAtm":
+            atm_list = _closest_atm(location)
+            ans_start = "Næsti hraðbanki"
+        elif result.qkey == "ClosestAtmDeposit":
+            atm_list = _closest_atm_deposit(location)
+            ans_start = "Næsti hraðbanki sem tekur við innborgunum"
+        elif result.qkey == "ClosestAtmForeignExchange":
+            atm_list = _closest_atm_foreign_exchange(location)
+        elif result.qkey == "ClosestAtmCoinmachine":
+            atm_list = _closest_atm_coinmachine(location)
```

Comment:
I cannot wait for the `match-case` statement introduced in Python 3.10!
Nice job keeping the `if-elif` statements compact and understandable :)
(They can become quite unwieldy in the query modules due to icelandic declensions)